crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -30,29 +30,29 @@ module sui::staking_pool {
 
     /// A staking pool embedded in each validator struct in the system state object.
     struct StakingPool has store {
+        // TODO: use object ID instead of validator address to identify a pool
         /// The sui address of the validator associated with this pool.
         validator_address: address,
         /// The epoch at which this pool started operating. Should be the epoch at which the validator became active.
         starting_epoch: u64,
         /// The total number of SUI tokens in this pool, including the SUI in the rewards_pool, as well as in all the principal
-        /// in the `Delegation` object, updated at epoch boundaries.
+        /// in the `StakedSui` object, updated at epoch boundaries.
         sui_balance: u64,
         /// The epoch delegation rewards will be added here at the end of each epoch. 
         rewards_pool: Balance<SUI>,
-        /// The number of delegation pool tokens we have issued so far. This number should equal the sum of
-        /// pool token balance in all the `Delegation` objects delegated to this pool. Updated at epoch boundaries.
-        delegation_token_supply: Supply<DelegationToken>,
-        /// Delegations requested during the current epoch. We will activate these delegation at the end of current epoch
-        /// and distribute staking pool tokens at the end-of-epoch exchange rate after the rewards for the current epoch
-        /// have been deposited.
-        pending_delegations: LinkedTable<ID, PendingDelegationEntry>,
+        /// Total number of pool tokens issued by the pool.
+        pool_token_balance: u64,
+        /// Exchange rate history of previous epochs. Key is the epoch number.
```

Comment:
- Might be useful to specify whether this is the exchange rate at the epoch beginning or end
- Might be useful to specify whether the lowest entry is at `starting_epoch` or something else.